torchdynamo/side_effects.py

```diff
@@ -300,7 +310,12 @@ def codegen(self, cg: PyCodegen):
                     cg.tx.output.update_co_names(name)
                     cg(value)
                     cg(var.mutable_local.source)
-                    suffixes.append([create_instruction("STORE_ATTR", name)])
+                    op = (
+                        "STORE_GLOBAL"
+                        if isinstance(var.mutable_local.source, GlobalSource)
```

Comment:
Will this work with (this code should work fine in the main branch)
```
some_global = Foo()

@torchdynamo.optimize(..)
def fn(x: Tensor):
  some_global.y = x + 1
```

I think in that case `var.mutable_local.source` will be `GlobalSource("some_global")`, so we will write to `global y` by mistake.

Perhaps rather than representing globals as
```
load_attr(gvar, gvar.source.name())
```
we should represent them as:
```
load_attr(global_scope_singleton, gvar.source.name())
```

Where global_scope_singleton is some sentinel object.  When the global is from some other module (because of inline), we can just map it to an attribute read directly.